src/components/Home/HomeMap.jsx

```diff
@@ -0,0 +1,141 @@
+        onClick={clicked}
+        onMouseEnter={mouseEnter}
+        onMouseLeave={mouseLeave}
+        color={color}
+        pointing='below'
+        circular
+      >
+        {text}
+      </Label>
+    )}
+    style={style}
+    size='mini'
+    on='click'
+    open={opened}
+    onClose={onClose}
+  >
+    <Menu.Header>
+      {city}
+    </Menu.Header>
+    <span>Posted: {date}</span>
+    <Menu size='tiny'>
+      <Menu.Item as='a'>
+        <Icon name='thumbs up' />
+        <Label color='teal' floating>
+          {upVote}
+        </Label>
+      </Menu.Item>
+      <Menu.Item as='a'>
+        <Icon name='thumbs down' />
+        <Label color='red' floating>
+          {downVote}
+        </Label>
+      </Menu.Item>
+    </Menu>
+  </Popup>
+);
+
+let today = new Date();
+let dateToday = today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate();
+
+class MapHome extends Component {
+  render(){
+    return(
+      <div style={{ height: '100%', width: '100%' }}>
+        {
+          !this.props.loading
+            ? (
+              <GoogleMapReact
+                // bootstrapURLKeys={{ key: /* YOUR KEY HERE */ }}
+                defaultCenter={this.props.userLocation}
+                defaultZoom={this.props.zoom}
+                onChange={this.props.changeBounds}
+              >
+                <AnyReactComponent
+                  lat={this.props.userLocation.lat}
+                  lng={this.props.userLocation.lng}
+                  date={dateToday}
+                  text="Current Location"
+                  city='Current Location'
+                />
+                {
+                  this.props.posts !== undefined
+                    ? (
+                      this.props.posts.map(post => {
+                        let markerColor = '';
+                        let markerClass = '';
+                        let popOpen = false;
+                        if (post.user_rating === 'Gnarly') {
+                          markerColor = 'red';
+                        } else if (post.user_rating === 'Good') {
+                          markerColor = 'orange';
+                        } else if (post.user_rating === 'Fair') {
+                          markerColor = 'yellow';
+                        } else if (post.user_rating === 'Poor') {
+                          markerColor = 'teal';
+                        } else if (post.user_rating === 'Flat') {
+                          markerColor = 'grey';
+                        }
+                        if (this.props.selectedPostId === post.id || this.props.clickedPostId === post.id) {
+                          markerClass = 'map-marker-active';
+                        }
+                        if (this.props.clickedPostId === post.id || this.props.clickedMarkerId === post.id) {
+                          popOpen = true;
+                        } else {
+                          popOpen = false;
+                        }
```

Comment:
So in React, NEVER do `if else...` statements OR declare variables inside `jsx`. You should try to only do ternary expressions (i.e. `isItTrue ? true : false`. Otherwise, place this as a component class method.